src/main/java/org/scijava/convert/DefaultConverter.java

```diff
@@ -168,10 +169,20 @@ public <T> T convert(final Object src, final Class<T> dest) {
 		}
 		if (saneDest == String.class) {
 			// destination type is String; use Object.toString() method
-			final String sValue = src.toString();
-			@SuppressWarnings("unchecked")
-			final T result = (T) sValue;
-			return result;
+			if (src.getClass().isArray()) {
+				final String elementString = ArrayUtils.toCollection(src).stream() //
+					.map(object -> convert(object, String.class)) //
+					.collect(Collectors.joining(","));
```

Comment:
Might be more human-readable to use `", "` (with trailing space) rather than `","`.

Reply:
Thanks, fixed.